# Transformada de Laplace inversa

Renato Naville Watanabe


Aqui vamos desenvolver um método para obter o sinal $x(t)$ a partir da sua transformada de Laplace. Esse procedimento é conhecido como **Transformada de Laplace Inversa**, denotada pelo operador $\mathcal{L}^{-1}$.

$\mathcal{L}^{-1}(X(s)) = x(t)$

A transformada de Laplace inversa é definida como:

$$x(t) = \frac{1}{2\pi j}\int_{c-j\infty}^{c+j\infty} X(s)e^{st}\,ds$$

Essa é uma integral no plano complexo. Felizmente, não precisaremos calcular essa integral. Nós usaremos os pares de sinais e transformadas já tabelados. Essa tabela é mostrada abaixo. Para sinais menos comuns, pode ser  necessário calcular essa integral. Se você estiver interessado nisso, pode procurar o livro de Brown e Churchill, listado nas referências, ou a disciplina *Funções de variáveis complexas* da UFABC.





Abaixo segue uma tabela com alguns sinais encontrados frequentemente na prática:

|$$\bf{x(t)}$$                        | $$\bf{X(s)}$$                                                          |
|-------------------------------------|------------------------------------------------------------------------|
|$\delta(t)$                          | 1                                                                      |
|$\delta(t-t_0)$                      | $e^{-st_0}$                                                            |
|$u(t)$                               | $\frac{1}{s}$                                                          |
|$u(t-t_0)$                           | $\frac{e^{-st}}{s}$                                                    |    
|$tu(t)$                              | $\frac{1}{s^2}$                                                        |
|$t^nu(t)$                            | $\frac{n!}{s^{n+1}}$                                                   |
|$e^{\lambda t}u(t)$                  | $\frac{1}{s-\lambda}$                                                  |
|$te^{\lambda t}u(t)$                 | $\frac{1}{(s-\lambda)^2}$                                              |
|$e^{-at}\cos(b t)u(t)$        |$\frac{s+a}{(s+a)^2+b^2}$                                                      |
|$e^{-at}\mathrm{sen}(b t)u(t)$|$\frac{b}{(s+a)^2+b^2}$                                                        |
|$e^{-at}\mathrm{cos}(b t+\theta)u(t)$|$\frac{(s+a)\cos(\theta) + b\mathrm{sen}(\theta)}{(s+a)^2+b^2}$         |     

Anteriormente, já vimos que para $x(t) = e^{-at}u(t)$, a transformada de Laplace é::

$X(s) = \frac{1}{s+a}$

Como estamos usando a transformada de Laplace unilateral, a relação entre o sinal $x(t)$ e a sua transformada de Laplace é única. Isso significa que se tivermos a transformada de Laplace $X(s) = \frac{1}{s+a}$, conseguimos obter o sinal $x(t)$ que gerou aquela transformada de Laplace:

$x(t) = e^{-at}u(t)$



---

## Exemplo 1

Considerando $X(s) = \frac{1}{s+0,3}$

A sua transformada de Laplace inversa é:

$x(t) = e^{-0.3t}u(t)$

---

---

## Exemplo 2

Considerando $X(s) = \frac{s}{s^2+9}$

A sua transformada de Laplace inversa é:

$x(t) = \cos(3t)u(t)$


---

Porém ao tentar fazer a transformada inversa de transformadas de Laplace como as mostradas abaixo, não conseguimos encontrar a sua transformada inversa diretamente.

$X_1(s) = \frac{s-3}{s^2 + 3s-3}$

$X_2(s) = \frac{s^2+0.2}{s^3 + 0.5s-1}$

$X_3(s) = \frac{s^2+0.2s+1}{s^4 +0.4s^3 - 0.3s^2-s}$

Estas transformadas de Laplace não têm um correspondente direto na tabela. Então para conseguirmos achar o sinal que gerou essas transformadas de Laplace precisamos reescrever a expressão de $X(s)$ em uma soma de expressões conhecidas. A maneira que iremos reescrever é a seguinte:

$$X(s) =  \frac{b_Ms^M +...+b_2s^2+b_1s + b_0}{s^N+ a_{N-1}s^{N-1}+...+a_1s+a_0} = A_0 + \frac{A_1}{s-p_1} + \frac{A_2}{s-p_2} + ... + \frac{A_N}{s-p_N}$$

em que $p_i$ são os polos de $X(s)$. O método que nos permite reescrever a expressão de $X(s)$ nesse formato é conhecido como **método das frações parciais**.

## Método das Frações Parciais

No método das frações parciais, primeiro reescrevemos a expressão de $X(z)$ no formato de polos e zeros.

$X(s) =\frac{b_Ms^M +...+b_2s^2+b_1s + b_0}{s^N+ a_{N-1}s^{N-1}+...+a_1s+a_0} = \frac{G(s-z_1)(s-z_2)...(s-z_M)}{(s-p_1)(s-p_2)...(s-p_N)}$

Queremos chegar no formato:

 $X(s) =  A_0 + \frac{A_1}{s-p_1} + \frac{A_2}{s-p_2} + ... + \frac{A_N}{s-p_N}$

### Caso em que $M<N$ com polos distintos

Vamos considerar primeiro o caso em que $M<N$. Nesse caso,

**1-** Então, igualamos as duas expressões:

$\frac{G(s-z_1)(s-z_2)...(s-z_M)}{(s-p_1)(s-p_2)...(s-p_N)}=A_0 + \frac{A_1}{s-p_1} + \frac{A_2}{s-p_2} + ... + \frac{A_N}{s-p_N}$

**2-** O que resta agora é encontrar os valores de $A_i$. Para isso, para cada $p_i$, multiplicamos os dois lados por $(s-p_i)$. Abaixo, para $p_1$.

$\frac{(s-p_1)G(z-z_1)(s-z_2)...(s-z_M)}{(s-p_1)(s-p_2)...(s-p_N)}=A_0(s-p_1) + \frac{A_1(s-p_1)}{s-p_1} + \frac{A_2(s-p_1)}{s-p_2} + ... + \frac{A_N(s-p_1)}{s-p_N}$

**3-** Perceba, que conseguimos cancelar os termos $(s-p_1)$ do lado esquerdo da igualdade e no termo sendo multiplicado por $A_1$.

$\frac{G(s-z_1)(s-z_2)...(s-z_M)}{(s-p_2)...(s-p_N)}=A_0(s-p_1) + A_1 + \frac{A_2(s-p_1)}{s-p_2} + ... + \frac{A_N(s-p_1)}{s-p_N}$

**4 -** Após isso, substituir $s$ pelo valor de $p_1$.

$\left.\frac{A(s-z_1)(s-z_2)...(s-z_M)}{(s-p_2)...(s-p_N)}\right|_{s=p_1} =\left.A_0(s-p_1)\right|_{s=p_1} + A_1 + \left.\frac{A_2(s-p_1)}{s-p_2}\right|_{s=p_1} + ... + \left.\frac{A_N(s-p_1)}{s-p_N}\right|_{s=p_1}$

**5-** Todos os termos do lado direito (com exceção do $A_1$) são zerados ao fazer essa substituição. Então, conseguimos encontrar o valor de $A_1$:

$A_1 = \frac{G(p_1-z_1)(p_1-z_2)...(p_1-z_M)}{(p_1-p_2)...(p_1-p_N)} $

**6-** Para encontrar o valor dos outros $A_i$, basta repetir os passos 2 até 5 para cada $p_i$. Para o caso de $M<N$, $A_0 = 0$.

**7-** Tendo encontrado todos os valores de $A_i$, agora é possível fazer a transformada de Laplace inversa de cada termo.



---

#### Exemplo 3

Conideramos aqui a transformada $X(s)$ dada pela seguinte expressão:

$$X(s) = \frac{s^2-0.3s-0.1}{s^3+0.2s^2-0.11s-0.012}$$

$X(s)$ pode ser reescrito da seguinte forma, com seus polos e zeros sendo mostrados explicitamente:

$$X(s) = \frac{(s-0.5)(s+0.2)}{(s+0.1)(s+0.4)(s-0.3)}$$

São 3 polos. Então,fazemos:

$\frac{(s-0.5)(s+0.2)}{(s+0.1)(s+0.4)(s-0.3)} = \frac{A_1}{s+0.1} + \frac{A_2}{s+0.4} + \frac{A_3}{s-0.3}$

Começando o procedimeto por $(s+0.1)$:

$\frac{(s+0.1)(s-0.5)(s+0.2)}{(s+0.1)(s+0.4)(s-0.3)} = \frac{A_1(s+0.1)}{s+0.1} + \frac{A_2(s+0.1)}{s+0.4} + \frac{A_3(s+0.1)}{s-0.3} \rightarrow \frac{(s-0.5)(s+0.2)}{(s+0.4)(s-0.3)} = A_1 + \frac{A_2(s+0.1)}{s+0.4} + \frac{A_3(s+0.1)}{s-0.3} $

Substituindo $s$ por $-0.1$:

$A_1 = \frac{(-0.1-0.5)(-0.1+0.2)}{(-0.1+0.4)(-0.1-0.3)} = 0.5$

Multiplicando agora por $(s+0.4)$:

$\frac{(s+0.4)(s-0.5)(s+0.2)}{(s+0.1)(s+0.4)(s-0.3)} = \frac{A_1(s+0.4)}{s+0.1} + \frac{A_2(s+0.4)}{s+0.4} + \frac{A_3(s+0.4)}{s-0.3} \rightarrow \frac{(s-0.5)(s+0.2)}{(s+0.1)(s-0.3)} = \frac{A_1(s+0.4)}{s+0.1} + A_2 + \frac{A_3(s+0.4)}{s-0.3} $

Substituindo $s$ por $-0.4$:

$A_2 = \frac{(-0.4-0.5)(-0.4+0.2)}{(-0.4+0.1)(-0.4-0.3)}=0.86$

Por fim, multiplicando por $(s-0.3)$:

$\frac{(s-0.3)(s-0.5)(s+0.2)}{(s+0.1)(s+0.4)(s-0.3)} = \frac{A_1(s-0.3)}{s+0.1} + \frac{A_2(s-0.3)}{s+0.4} + \frac{A_3(s-0.3)}{s-0.3} \rightarrow \frac{(s-0.5)(s+0.2)}{(s+0.1)(s+0.4)} = \frac{A_1(s-0.3)}{s+0.1} + \frac{A_2(s-0.3)}{s+0.4} + A_3$

Substituindo $s$ por $0.3$:

$A_3 = \frac{(0.3-0.5)(0.3+0.2)}{(0.3+0.1)(0.3+0.4)}=-0.36$

Então, podemos reescrever $X(s)$ como:

$X(s) = \frac{0.5}{s+0.1} + \frac{0.86}{s+0.4} - \frac{0.36}{s-0.3}$

Com isso, é fácil fazer a transformada inversa de cada um dos termos e recuperar o sinal $x(t)$:

$x(t) = 0.5e^{-0.1t}u(t) +  0.86e^{-0.4t}u(t) - 0.36e^{-0.3t}u(t)$

---

---

#### Exemplo 4

Considere a seguinte transformada de Laplace:

$X(s) = \frac{s+0.2}{s^3-0.5s^2+0.33s-0.029}$

$X(s)$ pode ser reescrito na forma de polos e zeros como:

$X(s) = \frac{s+0.2}{(s-0.1)(s-0.2 - j0.5)(s-0.2+j0.5)}$

Então temos que reescrever $X(z)$ como:

$\frac{s+0.2}{(s-0.1)(s-0.2 - j0.5)(s-0.2+j0.5)} = \frac{A_1}{s-0.1} + \frac{A_2}{s-0.2 - j0.5} + \frac{A_3}{s-0.2 + j0.5}$

Encontrando  o coeficiente $A_1$:

$A_1 = \left.\frac{s+0.2}{(s-0.2 - j0.5)(s-0.2+j0.5)}\right|_{s=0.1} = 1,15$

Agora, encontrando $A_2$:

$A_2 = \left.\frac{s+0.2}{(s-0.1)(s-0.2+j0.5)}\right|_{s=0.2+j0.5} = -0,58 - j1,12 = 1,26e^{-j2,05}$

e A_3:

$A_3 = \left.\frac{s+0.2}{(s-0.1)(s-0.2-j0.5)}\right|_{s=0.2-j0.5} = -0,58 + j1,12 = 1,26e^{j2,05}$

Repare que $A_3$ é o complexo conjugado de $A_2$. Isso sempre vai acontecer no caso de polos complexos.

Então, $X(s)$ é:

$X(s) = \frac{1,15}{s-0.1} + \frac{1,26e^{-j2,05}}{s-0.2 - j0.5} + \frac{1,26e^{j2,05}}{s-0.2 + j0.5}$

Então, a transformada de Laplace inversa de $X(s)$ é:

$x(t) = 1,15e^{0.1t}u(t) + 1,26e^{-j2,05}e^{(0.2+j0.5)t}u(t) + 1,26e^{j2,05}e^{(0.2-j0.5)t}u(t) = 1,15e^{0.1t}u(t) + 1,26e^{0.2t}e^{-j2,05}e^{j0.5t}u(t) + 1,26e^{0.2t}e^{j2,05}e^{-j0.5t}u(t)$

Aqui, talvez você possa achar que calculou os polos de $X(s)$ errado, já que $x(t)$ é uma expressão complexa. Mas repare que é possível usar a fórmula de Euler para obtermos cossenos. Sempre que aparecerem polos complexos isso será possível:

$x(t) =  1,15e^{0.1t}u(t) + 1,26e^{0.2t}(e^{j(0.5t-2,05)}+e^{-j(0.5t-2,05)})u(t)$

Pela fórmula de Euler:

$x(t) =  1,15e^{0.1t}u(t) + 2,52e^{0.2t}\cos(0.5t-2,05)u(t)$




---

### Caso em que $M<N$ com polos repetidos

Aqui, vamos supor que um dos polos seja reptido $K$ vezes. Caso mais de um polo seja repetido, o procedimento é o mesmo. Com o polo $p_1$ repetido $K$ vezes, temos $X(s)$ escrito como:

$X(s) = \frac{b_Ms^M +...+b_2s^2+b_1s + b_0}{s^N+ a_{N-1}s^{N-1}+...+a_1s+a_0} = \frac{G(s-z_1)(s-z_2)...(s-z_M)}{(s-p_1)^K(s-p_{K+1})...(s-p_N)}$

Aqui, não seria possível expandir $X(s)$ da mesma forma, por causa das raízes repetidas. Então, a expansão é feita para a seguinte forma:

$X(s) = \frac{B_1}{s-p_1} + \frac{B_2}{(s-p_1)^2} + ... + \frac{B_K}{(s-p_1)^K} + \frac{A_{K+1}}{s-p_{k+1}} + ... +  \frac{A_{M}}{s-p_{M}}$

Os coeficientes $A_i$, correspondentes aos polos não repetidos, são encontrados da mesma maneira que no caso anterior. Para encontrar os coeficientes $B_i$, fazemos o seguinte procedimento:

**1-** Multiplicamos as duas expressões por $(s-p_1)^K$ e cancelamos os termos que aparecem no numerador e denominador:

$\frac{A(s-z_1)(s-z_2)...(s-z_M)}{(s-p_{K+1})...(s-p_N)} = B_1(s-p_1)^{K-1} + B_2(s-p_1)^{K-2} + ... + B_K + \frac{A_{K+1}(s-p_1)^K}{s-p_{k+1}} + ... +  \frac{A_{M}(s-p_1)^K}{s-p_{M}}$

**2-** Substituindo $s$ por $p_1$, encontramos $B_K$:

$\left.\frac{A(s-z_1)(s-z_2)...(s-z_M)}{(s-p_{K+1})...(s-p_N)}\right|_{s=p_1} = \left.B_1(s-p_1)^{K-1}\right|_{s=p_1} + \left.B_2(s-p_1)^{K-2}\right|_{s=p_1} + ... + \left.B_K + \frac{A_{K+1}(s-p_1)^K}{s-p_{k+1}}\right|_{s=p_1} + ... +  \left.\frac{A_{M}(s-p_1)^K}{s-p_{M}}\right|_{s=p_1}$

$$B_K = \left.\frac{A(s-z_1)(s-z_2)...(s-z_M)}{(s-p_{K+1})...(s-p_N)}\right|_{s=p_1}$$

**3-** Para encontrar $B_{K-1}$, calculamos fazemos a derivada da expressão encontrada no passo 1.

$\frac{d}{ds}\left(\frac{A(s-z_1)(s-z_2)...(s-z_M)}{(s-p_{K+1})...(s-p_N)}\right) = \frac{d}{ds}\left(B_1(s-p_1)^{K-1}\right) + \frac{d}{ds}\left(B_2(s-p_1)^{K-2}\right) + ... + \frac{d}{ds}\left(B_{K-1}(s-p_1)\right) +\frac{d}{ds}\left(B_K\right) + \frac{d}{ds}\left(\frac{A_{K+1}(s-p_1)^K}{s-p_{k+1}}\right) + ... +  \frac{d}{ds}\left(\frac{A_{M}(s-p_1)^K}{s-p_{M}}\right)$

**4-** Isso faz com que o termo com $B_K$ desapareça e o termo $B_{K-1}$ fique sem estar multiplicando nada:

$\frac{d}{ds}\left(\frac{A(s-z_1)(s-z_2)...(s-z_M)}{(s-p_{K+1})...(s-p_N)}\right) = (K-1)B_1(s-p_1)^{K-2} + (K-2)B_2(s-p_1)^{K-3} + ... + B_{K-1} + \frac{d}{ds}\left(\frac{A_{K+1}(s-p_1)^K}{s-p_{k+1}}\right) + ... +  \frac{d}{ds}\left(\frac{A_{M}(s-p_1)^K}{s-p_{M}}\right)$

**5-** E fazemos de novo a substituição de s por $p_1$. Nessa etapa as derivadas com os coeficientes $A_i$ irão para zero:

$B_{K-1} = \left.\frac{d}{ds}\left(\frac{A(s-z_1)(s-z_2)...(s-z_M)}{(s-p_{K+1})...(s-p_N)}\right)\right|_{s=p_1}$

**6-** Para encontrar os $B_i$ restantes, basta continuar derivando a expressão encontrada no passo 1 e substituir $s$ por $p_1$.

---

#### Exemplo 5

Vamos considerar a seguinte transformada de Laplace:

$X(s) = \frac{(s-0.5)(s+0.3)}{(s+0.5)^2(s+0.1)}$

Para encontrar a sua transformada de Laplace inversa, primeiro vamos expandir em frações parciais:

$X(s) = \frac{A_1}{s+0.1} + \frac{B_1}{s+0.5} + \frac{B_2}{(s+0.5)^2}$

Primeiro, encontrando $A_1$:

$A_1 = \frac{(-0.1-0.5)(-0.1+0.3)}{(-0.1+0.5)^2} = -0.75$

Agora, $B_2$

$B_2 = \left.\frac{(s-0.5)(s+0.3)}{s+0.1}\right|_{s=_-0.5} = -0.5$

E para achar $B_1$:

$\left.\frac{d}{ds}\left(\frac{(s-0.5)(s+0.3)}{s+0.1} \right)\right|_{s=-0.5} = B_1$

$B_1 = \left.\frac{(2s-0.2)(s+0.1)-(s^2-0.2s-0.15)}{(s+0.1)^2}\right|_{s=-0.5} = 1.75$

Então, $X(s)$ pode ser escrito da seguinte forma:

$X(s) = \frac{-0.75}{s+0.1} + \frac{1.75}{s+0.5} - \frac{0.5}{(s+0.5)^2}$

Então, a transformada inversa é:

$x(t) = -0.75e^{-0.1t}u(t) + 1.75(e)^{-0.5t}u(t) - 0.5te^{-0.5t}u(t)$

---

### Caso com $M = N$

Caso o número de zeros seja igual o número de polos, antes de realizarmos o procedimento mostrado anteriormente, fazemos:

$X(s) = \frac{GN(s)}{D(s)} = G\frac{D(s) + N(s) - D(s)}{D(s)} = G + G\frac{N(s)-D(s)}{D(s)}$. Isso vai fazer com que o numerador fique com pelo menos um zero a menos. Após esse procedimento, é possível aplicar o método de frações parciais.

---

#### Exemplo 6

Considere a seguinte transformada de Laplace:

$X(s) = \frac{s^2+0.3s+0.02}{s^2+0.1s-0.56}$

Então, antes de iniciarmos o procedimento visto anteriormente, fazemos:

$X(s) = 1 +  \frac{s^2+0.3s+0.02 - (s^2+0.1s-0.56)}{s^2+0.1s-0.56} = 1 + \frac{0.2s+0.58}{s^2+0.1s-0.56}$

Agora, podemos expandir em frações parciais:

$X(s) = 1 + \frac{0.2s+0.58}{(s+0.8)(s-0.7)} = 1 + \frac{A_1}{s+0.8} + \frac{A_2}{s-0.7}$

Encontrando os coeficientes $A_1$ e $A_2$:

$A_1 = \left.\frac{0.2s+0.58}{(s-0.7)}\right|_{s=-0.8} = -0.28$

$A_2 = \left.\frac{0.2s+0.58}{(s+0.8)}\right|_{s=0.7}=0.48$


Então:

$x(t) = \delta(t) - 0.28e^{-0.8t}u(t) + 0.48e^{0.7t}u(t)$

---


## Caso $M> N$

Caso o número de zeros seja maior  do que o número de polos, antes de realizarmos o procedimento mostrado anteriormente, colocamos um fator do numerador em evidência, de forma que o polinômio restante no numerador fique com o mesmo grau do denominador. Após isso, realizamos o procedimento do caso anterior.

---

### Exemplo 7

Considere a seguinte transformada de Laplace:

$X(s) = \frac{s^3+0.3s^2+0.02s+1}{s^2+0.1s-0.56}$

Então, colocamos $s$ em evidência:

$X(s) = s\frac{s^2+0.3s+0.02+1/s}{s^2+0.1s-0.56} = s\frac{s^3+0.3s^2+0.02s+1}{s^3+0.1s^2-0.56s} = s\left(1 +s \frac{0.2s^2+0.58s+1}{s^3+0.1s^2-0.56s} \right) = s + \frac{0.2s^2+0.58s+1}{s^2+0.1s-0.56} = s + 0.2\frac{s^2+2.9s+5}{s^2+0.1s-0.56} = s + 0.2+\frac{2.8s+5.56}{s^2+0.1s-0.56} =s+0.2+\frac{2.8s+5.56}{(s+0.8)(s-0.7)} = s+ 0.2 + \frac{A_1}{s+0.8} + \frac{A_2}{s-0.7}$

$A_1 = \frac{(2,8.-0,8+5,56)}{-1.5}=-2,21$

$A_2 = \frac{(2,8.0,7+5,56)}{1.5}=5,01$

então

$x(t) = \frac{d\delta}{dt} + 0.2\delta(t) - 2,21e^{-0.8t}u(t) + 5,01e^{0.7t}u(t)$

---


In [ ]:
(2.8*0.7+5.56)/1.5

5.013333333333333

## Usando recursos computacionais

Realizar a expansão em frações parciais pode ser bastante trabalhoso. Hoje em dia, as linguagens usadas em computação científica têm maneiras para realizar essa expansão de forma simples. A biblioteca Scipy tem a função `residue`.

Basta colocar os coeficientes do numerador em uma variável, do denominador em outra variável e chamar a função residue. Chamando a função para a mesma transformada de Laplace do exemplo 6:

In [ ]:
from scipy.signal import residue
num = [1, 0.3, 0.02]
den = [1, 0.1, -0.56]
residue(num, den)

(array([ 0.48, -0.28]), array([ 0.7, -0.8]), array([1.]))

A função retorna 3 arrays (listas de números):

- A primeira lista tem os valores de $A_i$.

- A segunda a lista com os polos, na mesma ordem da lista anterior.

- A terceira lista com os valores que não correspondem a nenhum polo.

  Então, nesse caso a expansão fica:

  $X(s) = 1 + \frac{0.48}{s-0.7} - \frac{0.28}{s+0.8}$

  Que é o mesmo resultado encontrado no exemplo 6.

Outras linguagens, como Matlab, Julia e R, tem funções semelhantes.

---

## Exemplo 8

Vamos fazer a expansão em frações parciais de:

$X(s) = \frac{s^3-0.1s^2-0.17s-0.015}{s^4+1.3s^3+0.57s^2+0.095s+0.005}$



In [ ]:
num = [1, -0.1, -0.17,-0.015]
den = [1, 1.3,0.57,0.095, 0.005]

residue(num, den)

(array([-6.50521303e-16, -7.77777778e-01,  1.77777778e+00, -6.66666667e-01]),
 array([-0.1, -0.2, -0.5, -0.5]),
 array([], dtype=float64))

Perceba que a lista de polos deu dois polos repetidos. Nesse caso o segundo vai ser elevado ao quadrado.

Então, a expansão em frações parciais de $X(s)$ é:

$X(s) = -\frac{6.5.10^{-16}}{s+0.1} - \frac{0.777}{s+0.2}  + \frac{1.778}{s+0.5} - \frac{0.667}{(s+0.5)^2}$

O sinal que originou essa transformada de Laplace é:

$x(t) = -6.5.10^{-16}e^{-0.1t}u(t)-0.778e^{-0.2t}u(t) + 1.778e^{-0.5t}u(t) - 0.667te^{-0.5t}u(t)$

---

## Exercícios

**1)** Encontre os sinais que deram origem às seguintes transformadas de Laplace:

a) $X_1(s) = \frac{5}{(s+2)(s+3)}$


b) $X_2(s) = \frac{s+2}{s^2+4s+3}$


c) $X_3(s) = \frac{s+3}{s^2+4s+9}$

d) $X_4(s) = \frac{s+1}{s^2+4s+6}$



**2)** Encontre os sinais que deram origem às seguintes transformadas de Laplace:

a) $X_1(s) = \frac{e^{-2s}(2s+1)}{s^2+5s+4}$

b) $X_2(s) = \frac{se^{-2s}+3}{s^2+s+1}$

c) $X_3(s) = \frac{e^{-3s}+e^{-2s}+1}{s^2+2s+3}$

d) $X_4(s) = \frac{se^{-s}}{s^2+2s-3}$

**3)** Encontre os sinais que deram origem às seguintes transformadas de Laplace:

a) $X_1(s) = \frac{4}{(s+1)^3}$

b) $X_2(s) = \frac{(s+3)e^{-s}}{(s+4)^2}$

c) $x_3(s) = \frac{s+1}{s^2+4s+4}$

d) $x_4(s) = \frac{s-3}{s^2+3s+2,25}$

**4)** Encontre os sinais que deram origem às seguintes transformadas de Laplace:

a) $X_1(s) = \frac{4s^3-3s^2+5s+1}{(s+1)^3}$

b) $X_2(s) = \frac{s^2-2s-9}{s^2+4s+3}$

c) $X_3(s) = \frac{(s+1)^2}{s^2+4s+6}$

d) $x_4(s) = \frac{s^2-2s+3}{s^2+3s+2,25}$

## Referências

- Ulaby, FT e Yagle, AE, Signals & Systems: Theory and Applications, 2018, [ss2.eecs.umich.edu](ss2.eecs.umich.edu)
- Lathi, BP, Sinais e Sistemas Lineares, 2007
- Oppenheim, M.  e Willsky, S., Sistemas e Sinais, 2010
- Stewart, James. Cálculo - Volume 2, 2017
- Brown, James, e Ruel Churchill. Variáveis complexas e aplicações. (9ª edição), 2015.

## Resolução dos exercícios


**1)** Resolvido por Ivan Coqueiro

Encontre os sinais que deram origem às seguintes transformadas de Laplace:

**a) $X_1(s) = \frac{5}{(s+2)(s+3)}$**

\begin{align}
X_1(s) = \frac{5}{(s+2)(s+3)} ⇒ \\
X_1(s) = \frac{5}{(s+2)(s+3)}  = \frac{A_1}{(s+2)}  + \frac{A_2}{(s+3)}⇒ \\
X_1(s) = \frac{5}{(s+3)}  = A_1 + \frac{A_2(s+2)}{(s+3)}
\end{align}

Quando $s=-2$:

\begin{align}
    A_1 =5
\end{align}

Para a outra raíz:

\begin{align}
X_1(s) = \frac{5}{(s+2)(s+3)}  = \frac{A_1}{(s+2)}  + \frac{A_2}{(s+3)}⇒ \\
X_1(s) = \frac{5}{(s+2)}  = \frac{A_1(s+3)}{(s+2)} + A_2
\end{align}

Quando $s=-3$:

\begin{align}
    A_2 =-5
\end{align}

Portanto, por frações parciais:

\begin{align}
X_1(s) = \frac{5}{(s+2)}  - \frac{5}{(s+3)}
\end{align}

Realizando a transformada de Laplace inversa, obteremos:

\begin{align}
\mathcal{L}\{X_1(s)\} = \mathcal{L} \{\frac{5}{(s+2)}  - \frac{5}{(s+3)} \}\\ ⇒
x_1(t) = 5e^{-2t}u(t) - 5e^{-3t}u(t)
\end{align}




**b) $X_2(s) = \frac{s+2}{s^2+4s+3}$**

\begin{align}
X_2(s) = \frac{s+2}{s^2+4s+3} ⇒ \\
X_2(s) = \frac{s+2}{(s+1)(s+3)}  = \frac{A_1}{(s+1)}  + \frac{A_2}{(s+3)} ⇒ \\
X_2(s) = \frac{s+2}{(s+1)}  = \frac{A_1(s+3)}{(s+1)}  + A_2
\end{align}

Quando $s=-3$:

\begin{align}
 \frac{s+2}{(s+1)}  = \frac{A_1(s+3)}{(s+1)}  + A_2
\end{align}

\begin{align}
  A_2 = -0.5
\end{align}

Para $A_1$:

\begin{align}
X_2(s) = \frac{s+2}{(s+3)}  = A_1  + \frac{A_2(s+1)}{(s+3)}
\end{align}

Quando $s=-1$:

\begin{align}
 A_1  = 0.5
\end{align}

Convertendo para Transformada de Laplace Inversa:

\begin{align}
x_2(t) = 0.5e^{-t}u(t) - 0.5e^{-3t}u(t)
\end{align}

<p align="right"> $\square$.



**c) $X_3(s) = \frac{s+3}{s^2+4s+9}$**


\begin{align}
X_3(s) = \frac{s+3}{s^2+4s+9} ⇒ \\2
X_3(s) = \frac{s+3}{(s +2 - \sqrt{5}j)(s + 2 + \sqrt{5}j)}  = \frac{A_1}{s +2 - \sqrt{5}j}  + \frac{A_2}{(s + 2 + \sqrt{5}j)} ⇒ \\
X_3(s) = \frac{s+3}{(s +2 - \sqrt{5}j)(s +2 + \sqrt{5}j)} = A_1 + \frac{A_2(s +2 - \sqrt{5}j)}{(s +2 + \sqrt{5}j)}
\end{align}

Para $A_1$:

\begin{align}
 \frac{s+3}{(s +2 + \sqrt{5}j)} = A_1 + \frac{A_2(s +2 - \sqrt{5}j)}{(s +2 + \sqrt{5}j)}
\end{align}

Quando $s=-2 +\sqrt{5}j$:

\begin{align}
  A_1 = \frac{(-2+\sqrt{5}j)+3}{(-2+\sqrt{5}j+ 2+\sqrt{5}j)}  
\end{align}

\begin{align}
  A_1 = \frac{1+\sqrt{5}j}{2\sqrt{5}j}
\end{align}

Agora para $A_2$:

\begin{align}
\frac{s+3}{(s +2 - \sqrt{5}j)}  = \frac{A_1(s +2 + \sqrt{5}j)}{(s +2 - \sqrt{5}j)}  + A_2
\end{align}

Quando $s=-2 - \sqrt{5}j$:

\begin{align}
\frac{-2 - \sqrt{5}j+3}{(-2 - \sqrt{5}j +2 - \sqrt{5}j)}  = \frac{A_1(-2 - \sqrt{5}j +2 + \sqrt{5}j)}{(-2 - \sqrt{5}j +2 - \sqrt{5}j)}  + A_2
\end{align}

\begin{align}
 A_2  = -\frac{1 - \sqrt{5}j}{2\sqrt{5}j}
\end{align}

Convertendo para Transformada de Laplace Inversa:

\begin{align}
\frac{1+\sqrt{5}j}{2\sqrt{5}j}\frac{1}{s +2 - \sqrt{5}j}  -\frac{1 - \sqrt{5}j}{2\sqrt{5}j}\frac{1}{(s +2 + \sqrt{5}j)} ⇒\\
\end{align}

\begin{align}
x_3(t) = [\frac{1+\sqrt{5}j}{2\sqrt{5}j}e^{-(2 - \sqrt{5}j)t}  -\frac{1 - \sqrt{5}j}{2\sqrt{5}j}e^{-(2 + \sqrt{5}j)t}]u(t) ⇒\\
\end{align}


\begin{align}
x_3(t) = \frac{1}{2\sqrt{5}j}[(1+\sqrt{5}j)e^{-(2 - \sqrt{5}j)t}  -(1 - \sqrt{5}j)e^{-(2 + \sqrt{5}j)t}]u(t) ⇒\\
\end{align}


\begin{align}
x_3(t) = \frac{1}{2\sqrt{5}j}[(1+\sqrt{5}j)e^{-2t}e^{\sqrt{5}jt}  -(1 - \sqrt{5}j)e^{-2t}e^{-\sqrt{5}jt}]u(t) ⇒\\
\end{align}

\begin{align}
x_3(t) = e^{-2t}\left[\frac{(1+\sqrt{5}j)}{2\sqrt{5}j}e^{\sqrt{5}jt}  -\frac{(1 - \sqrt{5}j)}{2\sqrt{5}j}e^{-\sqrt{5}jt}\right]u(t) ⇒\\
\end{align}

\begin{align}
x_3(t) = e^{-2t}\left[0.55e^{-j0,42}e^{\sqrt{5}jt}  -0.55e^{j0.42}e^{-\sqrt{5}jt}\right]u(t) ⇒\\
\end{align}


Como $\cos{(x)} = \frac{e^{jx}+e^{-jx}}{2} $:

\begin{align}
x_3(t) = 1.1e^{-2t}\cos{(\sqrt{5}t-0.42)}u(t)
\end{align}


<p align="right"> $\square$.


**d) $X_4(s) = \frac{s+1}{s^2+4s+6}$**


\begin{align}
X_4(s) = \frac{s+1}{s^2+4s+6} ⇒ \\
\end{align}

\begin{align}
X_4(s) = \frac{s+1}{(s +2 + \sqrt{2}j)(s +2 - \sqrt{2}j)} ⇒ \\
\end{align}

Expandindo em frações parciais:

\begin{align}
 \frac{s+1}{(s +2 + \sqrt{2}j)(s +2 - \sqrt{2}j)} = \frac{A_1}{(s +2 + \sqrt{2}j)} + \frac{A_2}{(s +2 - \sqrt{2}j)}
\end{align}

Para $A_1$:

\begin{align}
 \frac{s+1}{(s +2 - \sqrt{2}j)} = A_1 + \frac{A_2(s +2 + \sqrt{2}j)}{(s +2 - \sqrt{2}j)}
\end{align}


Quando $s =  -2 - \sqrt{2}j$:

\begin{align}
 A_1 = \frac{-2 - \sqrt{2}j+1}{(-2 - \sqrt{2}j +2 - \sqrt{2}j)} ⇒ \\
 A_1 = \frac{-1- \sqrt{2}j}{(-2\sqrt{2}j)}
\end{align}
Agora para $A_2$:

\begin{align}
 \frac{s+1}{(s +2 + \sqrt{2}j)} = \frac{A_1(s +2 - \sqrt{2}j)}{(s +2 + \sqrt{2}j)} + A_2
\end{align}


Quando $s =  -2 + \sqrt{2}j$:

\begin{align}
 A_2 = \frac{-2 + \sqrt{2}j+1}{(-2 + \sqrt{2}j +2 + \sqrt{2}j)} = \frac{\sqrt{2}j-1}{2\sqrt{2}j}
\end{align}

Portanto, em expansão por frações parciais, teremos o seguinte resultado:

\begin{align}
X_4(s) = -(\frac{-\sqrt{2}j - 1}{2\sqrt{2}j}) \frac{1}{(s +2 + \sqrt{2}j)} + \frac{\sqrt{2}j-1}{2\sqrt{2}j}\frac{1}{(s +2 - \sqrt{2}j)}
\end{align}

Convertendo para Transformada de Laplace Inversa:

\begin{align}
x_4(t) = [-(\frac{-\sqrt{2}j - 1}{2\sqrt{2}j}) e^{-(2 + \sqrt{2}j)t} + \frac{\sqrt{2}j-1}{2\sqrt{2}j}e^{-(2 - \sqrt{2}j)t}]u(t) ⇒
\end{align}



\begin{align}
x_4(t) = \frac{e^{-2t}}{2\sqrt{2}j}[-e^{\sqrt{2}jt}+e^{-\sqrt{2}jt}+\sqrt{2}j(e^{\sqrt{2}jt} +e^{-\sqrt{2}jt}) ]u(t) ⇒
\end{align}


\begin{align}
x_4(t) = e^{-2t}[-\frac{(e^{\sqrt{2}jt}-e^{-\sqrt{2}jt})}{2\sqrt{2}j}+ \frac{(e^{\sqrt{2}jt} +e^{-\sqrt{2}jt})}{2} ]u(t)
\end{align}


Como $\sin{(x)} = \frac{e^{jx}-e^{-jx}}{2j}$ e $\cos{(x)} = \frac{e^{jx}+e^{-jx}}{2} $:


\begin{align}
x_4(t) = e^{-(2t)}[ -\frac{1}{\sqrt{2}}\sin{(\sqrt{2}t)} +\cos{(\sqrt{2}t)}]u(t)
\end{align}

<p align="right"> $\square$.


**3)** Resolvido por Ivan Coqueiro

a) $\frac{4}{(s+3)^3}$:

Temos um caso de raízes reais com pólos repetidos $(M<N):$ \\


Seguindo o passo-a-passo do professor (é possível ver um exercício similar no exemplo 5)

Temos a expressão por frações parciais dada abaixo:

$$\frac{4}{(s+1)^3} = \frac{A_1}{(s+1)} + \frac{A_2}{(s+1)^2} + \frac{A_3}{(s+1)^3}$$

**1-** Multiplicamos as duas expressões por $(s+1)^3$ e cancelamos os termos que aparecem no numerador e denominador:

$$4 = A_1(s+1)^2 + A_2(s+1) + A_3$$

**2-** Substituindo $s$ por $p_1=-1$, encontramos $A_3$:

$$A_3 = 4$$

**3-** Para encontrar $A_{2}$, calculamos fazemos a derivada da expressão encontrada no passo 1.

$\frac{d}{ds}\left(4\right) = \frac{d}{ds}\left(A_1(s+1)^2\right) + \frac{d}{ds}\left(A_2(s+1)\right) + \frac{d}{ds}\left(A_3\right)$

**4-** Isso faz com que o termo com $A_3$ desapareça e o termo $A_{2}$ fique sem estar multiplicando nada:

$0= \left(A_1\cdot2(s+1)\right) + \left(A_2\right) + 0$

**5-** E fazemos de novo a substituição de s por $p_1=-1$. Nessa etapa as derivadas com os coeficientes $A_i$ irão para zero:

$0= \left(A_1\cdot2(-1+1)\right) + \left(A_2\right) + 0$

$\Rightarrow A_2 = 0 $

**6-** Para encontrar os $A_1$, basta continuar derivando a expressão encontrada no passo 1 e substituir $s$ por $p_1$.

Assim teremos:
$A_1 = 0$

Por fim, realizando a transformação inversa da expressão:

$\mathcal{L^{-1}}\{ \frac{4}{(s+1)^3}\} ⇒ \frac{4}{2}t^2e^{-t}u(t)$

Logo teremos que:

$$x_1(t) = 2t^2e^{-t}u(t)$$

Podemos encontrar a expressão abreviada para esse tipo de caso, dada por:

$$A_{k-i} = \frac{1}{i!}\frac{d^i}{ds^i}(s+p_i)^kG(s) |_{s=-p_i}   ;  i=0,1,...,(k-1)$$

<p align="right"> $\square$.

b)$\frac{(s+3)e^{-s}}{(s+4)^2}$

Temos um caso de raízes reais com pólos repetidos $(M<N):$ \\


Seguindo o passo-a-passo do professor (é possível ver um exercício similar no exemplo 5)

Temos a expressão por frações parciais dada abaixo:

$$\frac{(s+3)e^{-s}}{(s+4)^2} = \frac{A_1}{(s+4)}e^{-s} + \frac{A_2}{(s+4)^2}e^{-s} $$

**1-** Multiplicamos as duas expressões por $(s+4)^2$ e cancelamos os termos que aparecem no numerador e denominador:

$$(s+3) = A_1(s+4) + A_2 $$

**2-** Substituindo $s$ por $p_1=-4$, encontramos $A_2$:

$$A_2 = -1$$

**3-** Para encontrar $A_{1}$, calculamos fazemos a derivada da expressão encontrada no passo 1.

$\frac{d}{ds}\left( (s+3) \right) = \frac{d}{ds}\left(A_1(s+4)\right) + \frac{d}{ds}\left(A_2\right)$

**4-** Isso faz com que o termo com $A_2$ desapareça e o termo $A_{1}$ fique sem estar multiplicando nada:

$1 = \left(A_1\right) +  0$



Por fim, realizando a transformação inversa da expressão:

$\mathcal{L^{-1}}\{ \frac{1}{(s+4)}e^{-s} - \frac{1}{(s+4)^2}e^{-s}\} ⇒ (e^{-4(t-1)}-(t-1)e^{-4(t-1)})u(t)$

Logo teremos que:

$$x_2(t) = (-e^{-4(t-1)})(t-2)u(t)$$

<p align="right"> $\square$.

c) $x_3(s) = \frac{s+1}{s^2+4s+4}$


A priori, é interessante decompor o Denominador D(s) nas raízes do polinômio. Obteremos os seguintes resultados:

$$x_3(s) = \frac{s+1}{s^2+4s+4}= \frac{s+1}{(s+2)^2}$$

Caso: polos reais e repetidos $(M<N)$. Teremos portanto a seguinte expressão em frações parciais:
$$ \frac{s+1}{(s+2)^2} = \frac{A_1}{(s+2)} + \frac{A_2}{(s+2)^2}$$

**1-** Multiplicamos as duas expressões por $(s+2)^2$ e cancelamos os termos que aparecem no numerador e denominador:

$$(s+1) = A_1(s+2) + A_2 $$

**2-** Substituindo $s$ por $p_1=-2$, encontramos $A_2$:

$$A_2 = -1$$

**3-** Para encontrar $A_{1}$, calculamos fazemos a derivada da expressão encontrada no passo 1.

$\frac{d}{ds}\left( (s+1) \right) = \frac{d}{ds}\left(A_1(s+2)\right) + \frac{d}{ds}\left(A_2\right)$

**4-** Isso faz com que o termo com $A_2$ desapareça e o termo $A_{1}$ fique sem estar multiplicando nada:

$$A_1 = 1$$


Por fim, realizando a transformação inversa da expressão:

$\mathcal{L^{-1}}\{ \frac{1}{(s+2)} - \frac{1}{(s+2)^2}\} ⇒ (e^{-2t} -te^{-2t})u(t) = -e^{-2t}(t-1)u(t)$

Logo teremos que:

$$x_3(t) = (e^{-2t} -te^{-2t})u(t)$$

<p align="right"> $\square$.

d) $x_4(s) = \frac{s-3}{s^2+3s+2,25}$

Para facilitar nesse caso, podemos utilizar o comando residue para encontrar as raízes do polinômio do denominador. O código abaixo mostra uma possível forma de resolver:

In [2]:
from scipy.signal import residue
num = [1, -3]
den = [1, 3, 2.25]

residue(num, den)

(array([ 1. , -4.5]), array([-1.5, -1.5]), array([], dtype=float64))

A expressão que procuramos é, portanto:

$$ \frac{s-3}{s^2+3s+2,25} = \frac{1}{(s+1,5)} - \frac{4,5}{(s+1,5)^2} $$

Realizando a transformação inversa da expressão:

$\mathcal{L^{-1}}\{ \frac{1}{(s+1,5)} - \frac{4,5}{(s+1,5)^2} \} ⇒ (e^{-1,5t} -4,5te^{-1,5t})u(t) = e^{-1,5t}(1-4,5t)u(t)$

Logo teremos que:

$$x_4(t) = (e^{-1,5t} -4,5te^{-1,5t})u(t)$$

<p align="right"> $\square$.